kano/window.py

```diff
@@ -134,6 +135,19 @@ def _get_window_by_id(wid):
     return gdk.window_foreign_new(int(wid))
 
 
+def retry_for(fn, secs, *args, **kwargs):
+    '''
+    Retry a function for some time if it returns None
+    '''
+    res = None
+    for i in range(1, secs * 10):
+        res = fn(*args, **kwargs)
+        if res is not None:
+            return res
+        time.sleep(0.1)
```

Comment:
Only concern is that the sleep time here (0.1) is highly dependant on the factor for the range, i.e. the 10 in `for i in range(1, secs * 10):`. Not really a problem but just a note.